enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,260 @@ any user-defined primary network.
+
+	// Subnet used inside the OVN network topology.
+	// E.g. 100.65.0.0/24. IPv6 (fd99::/64) and dual-stack (100.65.0.0/24,fd99::/64).
+	// When omitted and network `role` is `primary` the following default CIDR is used `100.65.0.0/24`, IPv6 `fd99::/64`.
+	// +optional
+	JoinSubnets []string `json:"joinSubnets,omitempty"`
+
+	// Control IP addresses management strategy. 
+	// When `Persistent` is specified it enable workloads have persistent IP addresses.
+	// For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).
+	// Supported by Topology Layer2 only.
+	// +kubebuilder:validation:Enum=Enable,Disable
+	// +optional
+	IPAMStrategy string `json:"ipamStrategy,omitempty"`
+}
+```
+- `Topology` and `Role` fields are mandatory.
+
+Suggested API validation rules:
+- `Topology` can be one of `Layer2`, `Layer3`.
+- `Role` can be one of `Primary`, `Secondary`.
+- `IPAMStrategy` can be `Persistent`.
+
+##### Cluster scoped CRD
+```golang
+// ClusterUserDefinedNetwork describes shared OVN network across namespaces request.
+type ClusterUserDefinedNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterUserDefinedNetworkSpec `json:"spec"`
+	// +optional
+	Status ClusterUserDefinedNetworkStatus `json:"status,omitempty"`
+}
+
+// ClusterUserDefinedNetwork defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetwork struct {
+	// NamespaceSelector Label selector for which namespace network should be available for.
+    // +kubebuilder:validation:Required
+    // +required
+	NamespaceSelector []metav1.LabelSelector `json:"namespaceSelector"`
+
+	// Template is direct specification of UserDefinedNetwork.
+	// +kubebuilder:validation:Required
+	// +required
+	Template *UserDefinedNetworkTemplateSpec `json:"template"`
+}
+
+// UserDefinedNetworkTemplateSpec UserDefinedNetwork spec template.
+type UserDefinedNetworkTemplateSpec struct {
+    // UserDefinedNetworkSpec contains the UserDefinedNetwork specification.
+	Spec UserDefinedNetworkSpec `json:"spec,omitempty"`
+}
+```
+
+###### Example
+Condition reflecting the network readiness:
+```yaml
+status:
+  conditions:
+  - type: NetworkReady
+    status: "True"
+    reason: NetworkAttachmentDefinitionReady
+    message: NetworkAttachmentDefinition has been created
+```
+Condition reflecting the network readiness underlying OVN network state:
+```yaml
+status:
+  conditions:
+  - type: NetworkCreated
+    status: "True"
+    reason: OVNNetworkCreated
+    message: OVN network has been created
+```
+
+##### Example - Namespace scoped network
+```yaml
+kind: UserDefinedNetwork
+metadata:
+  name: db-network
+  namespace: demo
+spec:
+  topology: Layer2
+  role: Primary
+  mtu: 9000
+  subnets: ["10.0.0.0/24"]
+  excludeSubnets: ["10.0.0.0/26"]
+  IPAMStrategy: Persistent
```

Comment:
We should not start with capital letters here.
```suggestion
  ipamLifecycle: Persistent
```

Reply:
Done